src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -737,32 +756,54 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref Compo
                     case TypeKind.Struct:
                     case TypeKind.Interface:
                     case TypeKind.Delegate:
-                        var typeArguments = ((NamedTypeSymbol)current).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
-                        if (typeArguments.IsEmpty)
+
+                        ImmutableArray<TypeWithAnnotations> typesWithAnnotations;
+
+                        if (current.IsAnonymousType)
+                        {
+                            var anonymous = (AnonymousTypeManager.AnonymousTypeOrDelegatePublicSymbol)current;
+                            var fields = anonymous.TypeDescriptor.Fields;
+                            var builder = ArrayBuilder<TypeWithAnnotations>.GetInstance();
+
+                            foreach (var field in fields)
+                            {
+                                builder.Add(field.TypeWithAnnotations);
+                            }
+
+                            typesWithAnnotations = builder.ToImmutableAndFree();
```

Comment:
>ToImmutableAndFree [](http://example.com/codeflow?start=59&length=18)

Is this allocation really necessary? It is probably better to avoid it by looping through ```fields```.

Reply:
Let me see.

Reply:
Updated. Please check with whitespace off.